File: RelayClient/src/de/twometer/iot/handler/util/PropertyMapper.java
package de.twometer.iot.handler.util;

import de.twometer.iot.bridge.ModeProperty;
import de.twometer.iot.bridge.Property;

import java.util.ArrayList;
import java.util.List;

public class PropertyMapper {

    private static List<Mapping> mappings = new ArrayList<>();

    static {
        map("Device.PowerState", "Alexa.PowerController", "powerState");
        map("Lamp.Brightness", "Alexa.BrightnessController", "brightness");
        map("Lamp.Color", "Alexa.ColorController", "color");
    }

    public static String getPropertyName(Property property) {
        if (property instanceof ModeProperty)
            return "mode";

        return findMapping(property).alexaProperty;
    }

    public static String getPropertyNamespace(Property property) {
        if (property instanceof ModeProperty)
            return "Alexa.ModeController";

        return findMapping(property).alexaNamespace;
    }

    private static Mapping findMapping(Property property) {
        for (Mapping mapping : mappings) {
            if (mapping.propertyName.equals(property.getName()))
                return mapping;
        }
        throw new IllegalArgumentException("Unknown property: " + property.getName());
    }

    private static void map(String propertyName, String controller, String mappedProp) {
        mappings.add(new Mapping(propertyName, controller, mappedProp));
    }

    private static class Mapping {
        private String propertyName;
        private String alexaNamespace;
        private String alexaProperty;

        public Mapping(String propertyName, String alexaNamespace, String alexaProperty) {
            this.propertyName = propertyName;
            this.alexaNamespace = alexaNamespace;
            this.alexaProperty = alexaProperty;
        }
    }

}

File: bridge/src/alexa/propertyMapper.js
'use strict';

const PropertyMap = [
    {
        alexaNamespace: 'Alexa.BrightnessController',
        alexaProperty: 'brightness',
        lumiProperty: 'Light.Brightness'
    },
    {
        alexaNamespace: 'Alexa.ColorTemperatureController',
        alexaProperty: 'colorTemperatureInKelvin',
        lumiProperty: 'Light.Temp'
    },
    {
        alexaNamespace: 'Alexa.ColorController',
        alexaProperty: 'color',
        lumiProperty: 'Light.Color'
    },
    {
        alexaNamespace: 'Alexa.PowerController',
        alexaProperty: 'powerState',
        lumiProperty: 'Device.PowerState'
    },
    {
        alexaNamespace: 'Alexa.PowerLevelController',
        alexaProperty: 'powerLevel',
        lumiProperty: 'Device.PowerLevel'
    },
    {
        alexaNamespace: 'Alexa.ModeController',
        alexaProperty: 'mode',
        lumiProperty: null
    }
];

module.exports = {

    getByNamespace(namespace) {
        return PropertyMap.filter(entry => entry.alexaNamespace === namespace)[0]
    },

    getByProperty(lumiName) {
        return PropertyMap.filter(entry => entry.lumiProperty === lumiName)[0]
    }
}